[torchrec] remove __getstate__ from PyTorch base optimizer (#16)

842974287 opened the pull request:
Summary:
Pull Request resolved: https://github.com/facebookresearch/torchrec/pull/16

`__getstate__` functions are used by pickle and the base class (PyTorch optimizer) overrides this function https://github.com/pytorch/pytorch/blob/6f29313e38881edeb8e877ed40626f56cb00c34a/torch/optim/optimizer.py#L56.

Torchrec optimizer classes has different attributes from the base class which is causing AttributeNotFound error during pickling.

This diff remove the `__getstate__` from the base class as it returns all the class members which is the default behavior and not need to override.

Test Plan:
PyTorch CI and torchrec unit test

buck test torchrec/optim/tests:test_keyed -- test_pickle

Differential Revision: D34147809

albanD reviewed (changes requested):
I am not sure why you need this to change? These are the fields for all the optimizers and this ensures we don't save things we don't care about?

Also you should be using state dict to serialize optimizers, why is this function important for your use case?

842974287 committed:
[torchrec] remove __getstate__ from PyTorch base optimizer (#72839)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/72839

Pull Request resolved: https://github.com/facebookresearch/torchrec/pull/16

`__getstate__` functions are used by pickle and the base class (PyTorch optimizer) overrides this function https://github.com/pytorch/pytorch/blob/6f29313e38881edeb8e877ed40626f56cb00c34a/torch/optim/optimizer.py#L56.

Torchrec optimizer classes has different attributes from the base class which is causing AttributeNotFound error during pickling.

This diff remove the `__getstate__` from the base class as it returns all the class members which is the default behavior and not need to override.

Test Plan:
PyTorch CI and torchrec unit test

buck test torchrec/optim/tests:test_keyed -- test_pickle

Differential Revision: D34147809

fbshipit-source-id: 2732fda8321699080a40d14803df70678abcf74a

842974287 commented:
@albanD Main reason I need to change it is that some torchrec optimizers don't have some of the fields specified here. Previously I thought we're returning all the class variables so it would be safe to use the default implementation. But I think if we are returning those on purpose then I should just change torchrec optimizers' behavior. I'll revert the changes here.

> Also you should be using state dict to serialize optimizers, why is this function important for your use case?

My use case is torch deploy and pickling was used to move python objects around.
